### PR TITLE
[automate-1877] clean up v2 in deployed envs and Cypress

### PR DESCRIPTION
**Pull request by bcmdarroch**
### :nut_and_bolt: Description: What code changed, and why?

Now that `upgrade-to-v2` upgrades Automate to `IAM v2.1`, all of our `v2`-specific testing is no longer needed. Automate users can now only be on v1 or v2.1; v2.0 doesn't exist any more on the customer-side

This PR and few follow-ups take care of removing anything `IAM v2.0`

### :+1: Definition of Done
- [x] remove acceptance/dev environments that were `v2`
- [x] replace `describeIfIAMV2` with `describeIfIAMV2p1` in Cypress tests

### :white_check_mark: Checklist

- [ ] Tests added/updated?
- [ ] Docs added/updated?

### :camera: Screenshots, if applicable

### CODE REVIEW

File: e2e/cypress/integration/ui/settings/02_team_mgmt.spec.ts [hunk 1]
```diff
@@ -69,8 +69,8 @@ describe('team management', () => {
 
               const systemTeams = ['admins', 'editors', 'viewers'];
               systemTeams.forEach(name => {
-              cy.get('#table-container chef-tr').contains(name)
-                .parent().parent().find('chef-control-menu').as('control-menu');
+                cy.get('#table-container chef-tr').contains(name)
```
Comment (bcmdarroch) [inline e2e/cypress/integration/ui/settings/02_team_mgmt.spec.ts]: used an auto-formatter on this file which caused a lot of little changes like this

File: e2e/cypress/integration/ui/settings/02_team_mgmt.spec.ts [hunk 2]
```diff
@@ -178,17 +178,18 @@ describe('team management', () => {
         cy.get('[data-cy=id-label]').contains(generatedTeamID);
 
         // initial state of dropdown
-        cy.get('app-team-management app-projects-dropdown #projects-selected').contains(unassigned);
+        cy.get('app-team-management app-projects-dropdown #projects-selected')
```
Comment (msorens) [inline e2e/cypress/integration/ui/settings/02_team_mgmt.spec.ts]: There are several more places in this function (like this) where you could use the introduced constant `projectDropdown`.

File: e2e/cypress/integration/api/iam/02_policies_api.spec.ts [hunk 3]
```diff
@@ -104,68 +107,72 @@ describeIfIAMV2('policies API', () => {
     });
   });
 
-  describeIfIAMV2p1('project assignment enforcement', () => {
+  describe('project assignment enforcement', () => {
```
Comment (msorens) [inline e2e/cypress/integration/api/iam/02_policies_api.spec.ts]: Is this OK because it is nested in a `describeIfIAMV2p1` above?

Comment (bcmdarroch) [inline e2e/cypress/integration/api/iam/02_policies_api.spec.ts]: this falls under the `describeIfIAMV2p1` block above

Comment (bcmdarroch) [reply e2e/cypress/integration/api/iam/02_policies_api.spec.ts]: yep!

**Pull request merged**